From f818dc11ab35e581e38123e51b57038693f30c81 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Fri, 5 Mar 2021 13:55:33 +0000
Subject: [PATCH] GCC: fix type confusion in surface_updater

GCC gets confused if an alias is declared with name of another
type, e.g. "using FeedStreamSurface = FeedStreamSurface". This can
be avoided by adding explicit class keyword.
---
 components/feed/core/v2/surface_updater.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/components/feed/core/v2/surface_updater.h b/components/feed/core/v2/surface_updater.h
index 42488a1..87a8e3e 100644
--- a/components/feed/core/v2/surface_updater.h
+++ b/components/feed/core/v2/surface_updater.h
@@ -27,7 +27,7 @@ class MetricsReporter;
 // changes (for spinners and zero-state).
 class SurfaceUpdater : public StreamModel::Observer {
  public:
-  using FeedStreamSurface = FeedStreamSurface;
+  using FeedStreamSurface = class FeedStreamSurface;
 
   explicit SurfaceUpdater(MetricsReporter* metrics_reporter);
   ~SurfaceUpdater() override;
-- 
2.26.2

